cmd/contour/ingressstatus.go

```diff
@@ -80,6 +84,12 @@ func (isw *loadBalancerStatusWriter) Start(ctx context.Context) error {
 		&networking_v1.Ingress{},
 	}
 
+	// Only create Gateway informer if a controller name was provided,
+	// otherwise the API may not exist in the cluster.
+	if len(isw.gatewayControllerName) > 0 {
```

Comment:
Would you get updates if the controller name wasn't defined?

Reply:
Without this check, we'd try to set up an informer for Gateways whether the user had configured Gateway API or not, but if the API wasn't even installed in the cluster, informer setup would fail.

Reply:
Oh that's right, there's a seperate informer which gets all the resources over again.